### ISSUE TITLE
Document deprecations

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#19806]
I noticed that Sphinx has a nice page in their docs documenting all the deprecations. https://www.sphinx-doc.org/en/master/extdev/deprecated.html

We should add the same. You can already see something similar to this by looking at the release notes, but it is hard to go through all the release notes to find deprecations, since most deprecations last for several releases.

**Comment by sylee957** [comment]
I'm also suggesting an idea of keeping track of changelog for each functions, classes, if the changes can be tracked by the diff automatically.
I rarely find the sympy release notes useful because there are just too many things to read.

**Comment by oscarbenjamin** [comment]
It's probably most useful to document deprecations in the docs for the functions/classes that are deprecated.

I think that especially with the removal of all of the docs for old versions we should have something more like core Python's `version added`, `version changed` etc notes.

**Comment by asmeurer** [comment]
I agree. Is there a special directive we can use for deprecations, or should we use `versionedadded`?

**Comment by oscarbenjamin** [comment]
There seems to be a "deprecated" directive:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated
"Similar to versionchanged, but describes when the feature was deprecated. An explanation can also be given, for example to inform the reader what should be used instead. Example:"
```
.. deprecated:: 3.1
   Use :func:`spam` instead.
```
